src/overmapbuffer.cpp

```diff
@@ -582,6 +582,13 @@ std::vector<tripoint> overmapbuffer::find_all( const tripoint& origin, const std
     return result;
 }
 
+tripoint overmapbuffer::find_random( const tripoint& origin, const std::string& type,
+                                     int dist, bool must_be_seen )
+{
+    std::vector<tripoint> result = find_all( origin, type, dist, must_be_seen );
+    return random_entry ( result, overmap::invalid_tripoint );
```

Comment:
Looks better when inlined: `random_entry( find_all( ... ), ...`. This way it doesn't depend on the type returned by `find_all()`. `find_random( ...., false )` is never invoked as well.